include/uart.h

```diff
@@ -64,6 +65,62 @@ extern "C" {
  */
 #define UART_ERROR_BREAK    (1 << 3)
 
+/**
+ * @brief UART controller configuration structure
+ *
+ * @param baudrate   Baudrate setting in bps.
+ * @param rx_timeout RX timeout in ms.
+ * @param flags      Bit field with following parts:
+ *
+ *     parity       [ 0 : 2 ]    - Parity bits.
+ *     stop_bits    [ 3 : 4 ]    - Stop bits.
+ *     data_bits    [ 5 : 7 ]    - Data bits.
+ *     flow_ctrl    [ 8 : 9 ]    - Flow control setting.
+ */
+struct uart_config {
+	u32_t baudrate;
+	u32_t rx_timeout;
+	u32_t flags;
```

Comment:
There is ongoing discussion if flow control, parity, stop, and data bits should be kept as bitmasks or enums.
@galak @tbursztyka 
I personally prefer using 4 values with enums here. Configuration will be kept inside driver, and this structure will only exist locally when it is used to extract current configuration, change it, then write it back.

Reply:
I'm for avoid bitfields for such things since I think it makes the code harder to get right.  I say go with u8_t's.

Reply:
4 values as 8bits instead of a bitfield in 32bits is same size so go for the enum.

Reply:
I prefer enum

Reply:
Changed to u8_t and enums

Reply:
Btw, so far I see discussion of bit**masks** vs byte-sized fields. Did we consider using true C bitfields, like `u32_t parity : 3`?

That would save the bits and allow address @erwango's comment below https://github.com/zephyrproject-rtos/zephyr/pull/11185/files#r231818904 (w/o increasing size).

Reply:
> we forbid long ago usage of C bitfields in public APIs

Really? LOL! Because we should do the opposite - forbid their use in referring to *external* structures (and we do that a lot using bitfields), and be absolutely free to use them in *our* APIs with *our* compilers. (Or with "their" compilers, as long as  we intend to keep "userspace and kernelspace is built together" paradigm.)

Reply:
That was the issue as far as I remember, app could be built by a different compiler than common zephyr one. This discussion was given almost 3 years ago.

Reply:
It was decided that C bitfields are forbidden and they will not be used in this API. I will stick with u8_t and enums.